e2e-tests/utils.js

```diff
@@ -23,21 +25,38 @@ class Utils {
      * @param {*} amount 
      * @param {*} nonce 
      */
-    static async signTransaction(from, to, amount, nonce){
-        const len = Compact.encodeU8a(Utils.TRANSACTION_LENGTH);
+    static signTransaction(from, to, amount, nonce){
         const tx = {
             from: u8aToHex(from.publicKey),
             to,
             amount,
             nonce
         };
-        let concatStr = "0x";
+        let concatStr = "";
         Object.values(tx).forEach(value => {
-            concatStr += value.slice(2);
+            concatStr += this._saveRemove0x(value.toString());
         });
-        const signature = from.sign(concatStr);
-        concatStr = u8aToHex(len) + concatStr.slice(2);
-        return concatStr.concat([u8aToHex(signature).slice(2) + "00"]);
+        const signature = from.sign(this._saveAdd0x(concatStr));
+        // add exhaustResources property at the end
+        concatStr = concatStr + this._saveRemove0x(u8aToHex(signature)) + "00";
+        const len = Compact.encodeU8a(concatStr.length/2);
+        concatStr = u8aToHex(len) + concatStr;
+        return this._saveAdd0x(concatStr);
+    }
+
+    /**
+     * Remove 0x prefix of Hex string
+     * @param {} str 
+     */
+    static _saveRemove0x(str){
```

Comment:
rename to `safe` not `save`. The term `safe` comes from the fact that you are safely removing 0x. Same thing for adding 0x (safely adding 0x)